### PR TITLE
SOE-2738: merged Caryl's changes

### PR DESCRIPTION
**Pull request by minorwm**
# READY FOR REVIEW

# Summary
- This adds two view blocks on the article pages for the info to be displayed for the collection it is apart of.
- This also changes the field_s_mag_article_collection in the Magazine article content type from "Unlimited" to "1" input value.

# Needed By (03/28/2018)
- Sprint end

# Criticality
- Client is eagerly awaiting this.

# Steps to Test
- Switch to this branch: https://github.com/SU-SOE/stanford_soe_helper/tree/7.x-2.x-collections-article-blocks
- Review code

Hey Bill, please update these steps to test for Greg and make sure they cover everything for the feature. Then, we can use these steps as a start for the Behat testing!!
Please update any changes in testing steps in this [ticket](https://stanfordits.atlassian.net/browse/SOE-2603)

## Verify feature
- Clear cache
 - Verify all Collection feature modules are in default state: `drush fd | grep collection`
## Add a new collection
- Add a new magazine collection at `node/add/stanford-collection`
- Verify the new magazine collection on the manage content page at `admin/manage/collection`
## Add articles to the collection
- Go to the manage content page for articles at `admin/manage/magazine_article`
- Click on `edit` for the articles you want to add to the new magazine collection (at least 18 articles)
- Flush cache
## Verify the collection page
- Visit your collection to verify the articles appear as expected (`email sign-up` and `load more` blocks should appear.
- Verify the `load more` button actually loads more blocks
## Verify the article page
- Click on an article in your collection
- Verify the items on the page: Top grey banner, Bottom block with 8 articles and a `see all` button
- Verify the `see all` button returns you to the collection page
- Verify that the `Magazine issue` block does _not_ appear on the page

# Affects 
- SoE (School of Engineering website)

# Associated Issues and/or People
## Related JIRA ticket(s)
https://stanfordits.atlassian.net/browse/SOE-2738
https://stanfordits.atlassian.net/browse/SOE-2739

## Folks to notify
@minorwm, @cjwest, @sherakama 

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: modules/stanford_soe_helper_collection_layouts/stanford_soe_helper_collection_layouts.info [hunk 1]
```diff
@@ -10,19 +10,28 @@ dependencies[] = ctools
...
+features[context][] = stanford_magazine_collection_article_engineering
 features[context][] = stanford_magazine_collection_engineering
 features[context][] = stanford_magazine_collection_list_page
 features[ctools][] = context:context:3
 features[ctools][] = strongarm:strongarm:1
 features[features_api][] = api:2
+features[field_base][] = field_s_mag_article_accent_color
+features[field_base][] = field_s_mag_article_dept
+features[field_instance][] = node-stanford_magazine_article-field_s_mag_article_accent_color
+features[field_instance][] = node-stanford_magazine_article-field_s_mag_article_dept
```
Comment (cjwest) [inline modules/stanford_soe_helper_collection_layouts/stanford_soe_helper_collection_layouts.info]: I think you will want to exclude these from the feature.

File: modules/stanford_soe_helper_collection_layouts/stanford_soe_helper_collection_layouts.info [hunk 2]
```diff
@@ -10,19 +10,28 @@ dependencies[] = ctools
 dependencies[] = features
 dependencies[] = image
 dependencies[] = pathauto
+dependencies[] = stanford_magazine
+dependencies[] = stanford_news_extras
+dependencies[] = stanford_people_spotlight
```
Comment (cjwest) [inline modules/stanford_soe_helper_collection_layouts/stanford_soe_helper_collection_layouts.info]: @minorwm, Are the dependencies on stanford_news_extras or stanford_people_spotlight needed?

File: modules/stanford_soe_helper_collection_views/stanford_soe_helper_collection_views.views_default.inc [hunk 3]
```diff
@@ -10,6 +10,356 @@
...
+  /* Field: Content: Subtitle */
+  $handler->display->display_options['fields']['field_s_collection_subtitle']['id'] = 'field_s_collection_subtitle';
+  $handler->display->display_options['fields']['field_s_collection_subtitle']['table'] = 'field_data_field_s_collection_subtitle';
+  $handler->display->display_options['fields']['field_s_collection_subtitle']['field'] = 'field_s_collection_subtitle';
+  $handler->display->display_options['fields']['field_s_collection_subtitle']['label'] = '';
+  $handler->display->display_options['fields']['field_s_collection_subtitle']['element_label_colon'] = FALSE;
+  /* Sort criterion: Content: Post date */
+  $handler->display->display_options['sorts']['created']['id'] = 'created';
+  $handler->display->display_options['sorts']['created']['table'] = 'node';
+  $handler->display->display_options['sorts']['created']['field'] = 'created';
```
Comment (cjwest) [inline modules/stanford_soe_helper_collection_views/stanford_soe_helper_collection_views.views_default.inc]: @minorwm 
Should this sort by the article publishing date instead? I think this is sorting by the collection creation date.

File: modules/stanford_soe_helper_collection_views/stanford_soe_helper_collection_views.views_default.inc [hunk 4]
```diff
@@ -10,6 +10,356 @@
...
+  /* Field: Content: Featured Image */
+  $handler->display->display_options['fields']['field_s_mag_article_image']['id'] = 'field_s_mag_article_image';
+  $handler->display->display_options['fields']['field_s_mag_article_image']['table'] = 'field_data_field_s_mag_article_image';
+  $handler->display->display_options['fields']['field_s_mag_article_image']['field'] = 'field_s_mag_article_image';
+  $handler->display->display_options['fields']['field_s_mag_article_image']['label'] = '';
+  $handler->display->display_options['fields']['field_s_mag_article_image']['exclude'] = TRUE;
+  $handler->display->display_options['fields']['field_s_mag_article_image']['element_label_colon'] = FALSE;
+  $handler->display->display_options['fields']['field_s_mag_article_image']['click_sort_column'] = 'fid';
+  $handler->display->display_options['fields']['field_s_mag_article_image']['settings'] = array(
+    'image_style' => '6-col-banner-short',
```
Comment (cjwest) [inline modules/stanford_soe_helper_collection_views/stanford_soe_helper_collection_views.views_default.inc:168]: @minorwm This image style, 6-col-banner-short, is in Stanford Image Styles.  Make sure you've added that module as a dependency.

File: scss/components/_soe_dm_collection.scss [hunk 5]
```diff
@@ -133,6 +133,58 @@
...
+// A DIGITAL MAGAZINE ARTICLE COLLECTION BLOCKS
+//
+
+// A ARTICLE COLLECTION BLOCK
+
+.node-type-stanford-magazine-article {
+  .view-stanford-magazine-article-collection {
+    padding-bottom: 40px;
+    padding-top: 40px;
+    background-color: #e4e4e4;
```
Comment (cjwest) [inline scss/components/_soe_dm_collection.scss]: Please check with @kerri-augenstein , we should use the [pre-defined variables](https://github.com/SU-SOE/stanford_soe_helper/blob/7.x-2.x/scss/variables/_soe_colors.scss). I'm thinking this should be $light-gray

File: modules/stanford_soe_helper_collection_views/stanford_soe_helper_collection_views.views_default.inc (hunk 4 above)
Comment (minorwm) [reply modules/stanford_soe_helper_collection_views/stanford_soe_helper_collection_views.views_default.inc:168]: @cjwest I have it as a dependency

**Comment by minorwm** [comment]
@cjwest This is ready for review again.

File: modules/stanford_soe_helper_collection_views/stanford_soe_helper_collection_views.views_default.inc (hunk 4 above)
Comment (cjwest) [reply modules/stanford_soe_helper_collection_views/stanford_soe_helper_collection_views.views_default.inc:168]: 👍

File: modules/stanford_soe_helper_collection_layouts/stanford_soe_helper_collection_layouts.info [hunk 6]
```diff
@@ -7,22 +7,34 @@ project = stanford_soe_helper_collection_layouts
...
 dependencies[] = context
 dependencies[] = ctools
+dependencies[] = ds
 dependencies[] = features
 dependencies[] = image
+dependencies[] = options
 dependencies[] = pathauto
+dependencies[] = stanford_magazine
+dependencies[] = stanford_news_extras
+dependencies[] = stanford_people_spotlight
```
Comment (cjwest) [inline modules/stanford_soe_helper_collection_layouts/stanford_soe_helper_collection_layouts.info]: @minorwm Do we need stanford_news_extras and stanford_people_spotlight as dependencies?

File: modules/stanford_soe_helper_collection_layouts/stanford_soe_helper_collection_layouts.info [hunk 7]
```diff
@@ -7,22 +7,34 @@ project = stanford_soe_helper_collection_layouts
...
+dependencies[] = taxonomy
 dependencies[] = views
 dependencies[] = xmlsitemap
+features[context][] = stanford_magazine_collection_article_engineering
 features[context][] = stanford_magazine_collection_engineering
 features[context][] = stanford_magazine_collection_list_page
 features[ctools][] = context:context:3
 features[ctools][] = strongarm:strongarm:1
 features[features_api][] = api:2
 features[variable][] = pathauto_node_stanford_collection_pattern
```
Comment (cjwest) [inline modules/stanford_soe_helper_collection_layouts/stanford_soe_helper_collection_layouts.info:29]: @minorwm Just checking: do you need this variable: pathauto_node_stanford_collection_pattern

Comment (minorwm) [reply modules/stanford_soe_helper_collection_layouts/stanford_soe_helper_collection_layouts.info:29]: @cjwest this is needed

**Comment by pookmish** [comment]
👍 from code review standpoint

**Pull request merged**